src/SymphonyOSS.RestApiClient/Generated/OpenApi/AgentApi/AgentApi.cs

```diff
@@ -4519,46 +4529,79 @@ public string BaseUrl
                     request_.Content = content_;
                     request_.Method = new System.Net.Http.HttpMethod("POST");
                     request_.Headers.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
-    
+
                     PrepareRequest(client_, request_, urlBuilder_);
                     var url_ = urlBuilder_.ToString();
                     request_.RequestUri = new System.Uri(url_, System.UriKind.RelativeOrAbsolute);
                     PrepareRequest(client_, request_, url_);
-    
+
+                    var requestDetails = new Dictionary<string, string>()
+                    {
+                        {"url", url_}
+                    };
+                    _log.LogDebug("Attempting to send the request to symphony.", requestDetails);
+
                     var response_ = await client_.SendAsync(request_, System.Net.Http.HttpCompletionOption.ResponseHeadersRead, cancellationToken).ConfigureAwait(false);
+
+                    _log.LogDebug("Request successfully sent to symphony and a response was received.");
+
                     try
                     {
+                        _log.LogDebug("Attempting to process the response from symphony.");
                         var headers_ = System.Linq.Enumerable.ToDictionary(response_.Headers, h_ => h_.Key, h_ => h_.Value);
                         foreach (var item_ in response_.Content.Headers)
                             headers_[item_.Key] = item_.Value;
-    
+
                         ProcessResponse(client_, response_);
-    
-                        var status_ = ((int)response_.StatusCode).ToString();
+
+                        _log.LogDebug("Successfully processed the response from symphony.");
+
+                        var status_ = ((int) response_.StatusCode).ToString();
+
+                        var statusDetails = new Dictionary<string, string>()
+                        {
+                            {"status", status_}
+                        };
+
+                        var responseDataDetails = new Dictionary<string, string>()
```

Comment:
why cant we do this after we have the responseData ?

Reply:
Ah I see why now.

Reply:
Then we would be creating a new object for every if case.

Reply:
do we really need a dictionary here if the responseData is always a string?